src/main/java/hudson/plugins/ansicolor/AnsiColorBuildWrapper.java

```diff
@@ -117,19 +115,139 @@ private void addAll(AnsiColorMap[] maps, Map<String, AnsiColorMap> to) {
         @Override
         public boolean configure(final StaplerRequest req, final JSONObject formData) throws FormException {
             try {
-                setColorMaps(req.bindJSONToList(AnsiColorMap.class,
-                        req.getSubmittedForm().get("colorMap")).toArray(new AnsiColorMap[1]));
-                setGlobalColorMapName(req.getSubmittedForm().getString("globalColorMapName"));
+                final List<AnsiColorMap> colorMaps = req.bindJSONToList(AnsiColorMap.class, req.getSubmittedForm().get("colorMap"));
+                for (AnsiColorMap colorMap : colorMaps) {
+                    validateFieldName(colorMap.getName());
+                    validateFieldColorLiteral(colorMap.getBlack(), "black");
```

Comment:
I would swap value and name, feels weird that it's value, then name, but no big deal.

Maybe there's a way to loop over these?

Reply:
> I would swap value and name

I started off with the value alone and only later added the field name. Maybe it does make more sense to have the name on the first spot.

> Maybe there's a way to loop over these?

There's not much that can be done here I'm afraid. Maybe putting the field names and method references in a map and looping through that. Do you think this approach looks better?

Reply:
I like what you did to this plus I would extract that map in a method.

Reply:
Moved it to a static Map.